lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -2253,7 +2253,8 @@ bool postProcessBuiltinReturningStruct(Function *F) {
       NewCI->setCallingConv(CI->getCallingConv());
       SmallVector<User *> CIUsers(CI->users());
       for (auto *CIUser : CIUsers) {
-        CIUser->replaceUsesOfWith(CI, A);
+        CIUser->replaceUsesOfWith(CI,
```

Comment:
This would be a little awkward, but we have to check if the user is Load. If it's Load we should either:
- not create a new Load;
- remove the old load
Because otherwise...

Reply:
good point, thank you

Reply:
Updated, please have a look. The point was not in load user, but rather in that (1) we shouldn't update StoreInst that is going to be removed further, (2) it's enough to have one load per all users.